exporters/metric/cortex/auth.go

```diff
@@ -88,3 +92,99 @@ func (e *Exporter) addBearerTokenAuth(req *http.Request) error {
 
 	return nil
 }
+
+// buildClient returns a http client that uses TLS and has the user-specified proxy and
+// timeout.
+func (e *Exporter) buildClient() (*http.Client, error) {
+	// Create a TLS Config struct for use in a custom HTTP Transport.
+	tlsConfig, err := e.buildTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+
+	// Convert proxy url to proxy function for use in the created Transport.
+	if e.config.ProxyURL != "" {
+		proxyURL, err := url.Parse(e.config.ProxyURL)
```

Comment:
Can we switch the config to store the `ProxyURL` field as a `url.URL` to avoid this parse here? Seems like this should be part of the config validation.